client/signup/steps/woocommerce-install/transfer/index.tsx

```diff
@@ -14,6 +14,16 @@ export default function Transfer( props: WooCommerceInstallProps ): ReactElement
 	const siteId = useSelector( getSelectedSiteId ) as number;
 	const isAtomic = useSelector( ( state ) => isAtomicSite( state, siteId ) );
 
+	const {
+		goToStep,
+		signupDependencies: { siteConfirmed },
+	} = props;
+
+	if ( siteConfirmed !== siteId ) {
```

Comment:
Checking this isn't *that* important in the end (dispatching `ubmitSignupStep` was the real fix) but because state gets stored I want to rule out any potential resume problems.